src/Mvc/Collection.php

```diff
@@ -0,0 +1,1555 @@
+<?php
+
+/** @noinspection PhpUnused */
+
+/**
+ * This file is part of the Phalcon Framework.
+ *
+ * (c) Phalcon Team <team@phalconphp.com>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Phalcon\Incubator\Mvc;
+
+use JsonSerializable;
+use MongoDB\BSON\ObjectId;
+use MongoDB\BSON\Serializable as BsonSerializable;
+use MongoDB\BSON\Unserializable;
+use MongoDB\Database;
+use MongoDB\Driver\Cursor;
+use Phalcon\Di;
+use Phalcon\Di\AbstractInjectionAware;
+use Phalcon\Di\DiInterface;
+use Phalcon\Events\ManagerInterface as EventsManagerInterface;
+use Phalcon\Helper\Str;
+use Phalcon\Incubator\Mvc\Collection\BehaviorInterface;
+use Phalcon\Incubator\Mvc\Collection\Exception;
+use Phalcon\Incubator\Mvc\Collection\ManagerInterface;
+use Phalcon\Messages\Message;
+use Phalcon\Messages\MessageInterface;
+use Phalcon\Mvc\EntityInterface;
+use Phalcon\Validation\ValidationInterface;
+use Serializable;
+
+/**
+ * Class Collection
+ *
+ * ActiveRecord class for the management of MongoDB collections
+ *
+ * @package Phalconator\Mvc
+ */
+class Collection extends AbstractInjectionAware implements
+    CollectionInterface,
+    EntityInterface,
+    Serializable,
+    BsonSerializable,
+    Unserializable,
+    JsonSerializable
+{
+    public const DIRTY_STATE_PERSISTENT = 0;
+    public const DIRTY_STATE_TRANSIENT = 1;
+    public const DIRTY_STATE_DETACHED = 2;
+
+    public const OP_NONE = 0;
+    public const OP_CREATE = 1;
+    public const OP_UPDATE = 2;
+    public const OP_DELETE = 3;
+
+    /**
+     * @var ObjectId|mixed $_id
+     */
+    protected $_id;
+
+    /**
+     * @var Database|mixed
+     */
+    protected $connection;
+
+    /**
+     * @var DiInterface|null
+     */
+    protected $container;
+
+    protected $dirtyState = 1;
+
+    protected static $disableEvents = false;
+
+    /**
+     * @var array $reserved
+     */
+    protected static $reserved;
+
+    /**
+     * @var array $typeMap
+     */
+    protected static $typeMap;
+
+    protected $errorMessages = [];
+
+    /**
+     * @var ManagerInterface|null
+     */
+    protected $collectionsManager;
+
+    protected $operationMade = 0;
+
+    protected $skipped = false;
+
+    /**
+     * Collection constructor.
+     *
+     * @param null $data
+     * @param DiInterface|null $container
+     * @param ManagerInterface|null $collectionsManager
+     * @throws Exception
+     */
+    final public function __construct(
+        $data = null,
```

Comment:
```suggestion
        array $data = [],
```

Reply:
https://github.com/phalcon/cphalcon/blob/master/phalcon/Mvc/Model.zep#L138
I followed the syntax of the Phalcon models. Should I change?